File: apps/android/feature/exercises/ui/src/main/kotlin/com/scribblefit/feature/exercises/ui/components/TrendsSection.kt
package com.scribblefit.feature.exercises.ui.components

import androidx.compose.foundation.background
import androidx.compose.foundation.layout.Arrangement
import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.Row
import androidx.compose.foundation.layout.Spacer
import androidx.compose.foundation.layout.fillMaxWidth
import androidx.compose.foundation.layout.padding
import androidx.compose.foundation.layout.size
import androidx.compose.foundation.layout.width
import androidx.compose.foundation.shape.CircleShape
import androidx.compose.material.icons.Icons
import androidx.compose.material.icons.automirrored.filled.KeyboardArrowRight
import androidx.compose.material3.Icon
import androidx.compose.material3.Text
import androidx.compose.material3.TextButton
import androidx.compose.runtime.Composable
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp
import com.scribblefit.core.designsystem.ScribbleFitTheme
import com.scribblefit.core.model.ExerciseTrends
import com.scribblefit.core.model.TrendDirection

@Composable
fun TrendsSection(
    trends: ExerciseTrends,
    weightUnit: String,
    titleLabel: String,
    viewAllLabel: String,
    current1rmLabel: String,
    intensityLabel: String,
    weightVsLastLabel: String,
    lastVolumeLabel: String,
    onViewAllClick: () -> Unit,
    getTrendDirectionText: @Composable (TrendDirection) -> String,
    modifier: Modifier = Modifier
) {
    Column(modifier = modifier.fillMaxWidth()) {
        Row(
            modifier = Modifier.fillMaxWidth(),
            horizontalArrangement = Arrangement.SpaceBetween,
            verticalAlignment = Alignment.CenterVertically
        ) {
            Text(
                text = titleLabel,
                style = ScribbleFitTheme.typography.labelSmall,
                color = ScribbleFitTheme.colors.midGray,
                fontWeight = FontWeight.Bold,
                letterSpacing = 1.sp
            )

            TextButton(onClick = onViewAllClick) {
                Text(
                    text = viewAllLabel,
                    style = ScribbleFitTheme.typography.labelSmall,
                    color = ScribbleFitTheme.colors.primary,
                    fontWeight = FontWeight.Bold
                )
                Icon(
                    imageVector = Icons.AutoMirrored.Filled.KeyboardArrowRight,
                    contentDescription = null,
                    tint = ScribbleFitTheme.colors.primary,
                    modifier = Modifier.size(16.dp)
                )
            }
        }

        Column(
            modifier = Modifier.padding(top = ScribbleFitTheme.spacing.small),
            verticalArrangement = Arrangement.spacedBy(ScribbleFitTheme.spacing.smallLarger)
        ) {
            TrendItem(
                label = current1rmLabel,
                value = "${trends.estimated1RM.toInt()}$weightUnit",
                direction = trends.trendDirection,
                getTrendDirectionText = getTrendDirectionText
            )

            TrendItem(
                label = intensityLabel,
                value = "${(trends.intensity * 100).toInt()}%",
                direction = TrendDirection.STABLE,
                getTrendDirectionText = getTrendDirectionText
            )

            TrendItem(
                label = weightVsLastLabel,
                value = "${if (trends.improvement >= 0) "+" else ""}${(trends.improvement * 100).toInt()}%",
                direction = trends.trendDirection,
                getTrendDirectionText = getTrendDirectionText
            )

            TrendItem(
                label = lastVolumeLabel,
                value = "${trends.lastVolume.toInt()}$weightUnit",
                direction = trends.lastVolumeTrend,
                getTrendDirectionText = getTrendDirectionText
            )
        }
    }
}

@Composable
private fun TrendItem(
    label: String,
    value: String,
    direction: TrendDirection,
    getTrendDirectionText: @Composable (TrendDirection) -> String
) {
    Row(
        modifier = Modifier.fillMaxWidth(),
        verticalAlignment = Alignment.CenterVertically,
        horizontalArrangement = Arrangement.SpaceBetween
    ) {
        Column {
            Text(
                text = label,
                style = ScribbleFitTheme.typography.bodyMedium,
                color = ScribbleFitTheme.colors.primary,
                fontWeight = FontWeight.Medium
            )
        }

        Row(verticalAlignment = Alignment.CenterVertically) {
            Text(
                text = value,
                style = ScribbleFitTheme.typography.bodyLarge,
                color = ScribbleFitTheme.colors.primary,
                fontWeight = FontWeight.Bold
            )

            Spacer(modifier = Modifier.width(ScribbleFitTheme.spacing.smallLarger))

            TrendBadge(
                direction = direction,
                text = getTrendDirectionText(direction)
            )
        }
    }
}

@Composable
private fun TrendBadge(direction: TrendDirection, text: String) {
    val color = when (direction) {
        TrendDirection.IMPROVING -> ScribbleFitTheme.colors.successGreen
        TrendDirection.STABLE -> ScribbleFitTheme.colors.midGray
        TrendDirection.PLATEAUED -> ScribbleFitTheme.colors.warningOrange
        TrendDirection.DECLINING -> ScribbleFitTheme.colors.dangerRed
    }

    Text(
        text = text,
        style = ScribbleFitTheme.typography.labelSmall,
        color = color,
        fontWeight = FontWeight.Bold,
        modifier = Modifier
            .background(color.copy(alpha = 0.1f), CircleShape)
            .padding(horizontal = 10.dp, vertical = 4.dp)
    )
}

File: apps/ios/ScribbleFit/LocalPackages/Features/FeatureExercises/Sources/FeatureExercises/UI/Components/TrendsSection.swift
import SwiftUI
import CoreModel
import CoreDesignSystem

struct TrendsSection: View {
    let trends: ExerciseTrends
    let weightUnit: String
    let onViewAllClick: () -> Void

    var body: some View {
        VStack(alignment: .leading, spacing: 16) {
            HStack {
                Text(String(localized: "TRENDS"))
                    .font(.scribbleLabelMedium)
                    .fontWeight(.bold)
                    .kerning(1)
                    .foregroundStyle(Color.scribbleMidGray)

                Spacer()

                Button(action: onViewAllClick) {
                    HStack(spacing: 4) {
                        Text(String(localized: "VIEW ALL"))
                            .font(.scribbleLabelMedium)
                            .fontWeight(.bold)

                        Image(systemName: "chevron.right")
                            .font(.system(size: 10, weight: .bold))
                    }
                    .foregroundStyle(Color.scribblePrimary)
                }
            }

            VStack(spacing: 20) {
                TrendItem(
                    label: String(localized: "Current 1RM"),
                    value: "\(Int(trends.current1RM))\(weightUnit)",
                    direction: trends.trendDirection
                )

                TrendItem(
                    label: String(localized: "Intensity"),
                    value: "\(Int(trends.intensity * 100))%",
                    direction: .stable // Intensity doesn't have a trend direction in this view
                )

                TrendItem(
                    label: String(localized: "Weight vs Last"),
                    value: "\(trends.improvement >= 0 ? "+" : "")\(Int(trends.improvement * 100))%",
                    direction: trends.trendDirection
                )

                TrendItem(
                    label: String(localized: "Last Volume"),
                    value: "\(Int(trends.lastVolume))\(weightUnit)",
                    direction: trends.lastVolumeTrend
                )
            }
        }
        .frame(maxWidth: .infinity, alignment: .leading)
    }
}

private struct TrendItem: View {
    let label: String
    let value: String
    let direction: TrendDirection

    var body: some View {
        HStack {
            Text(label)
                .font(.scribbleBodyMedium)
                .fontWeight(.medium)
                .foregroundStyle(Color.scribblePrimary)

            Spacer()

            HStack(spacing: 12) {
                Text(value)
                    .font(.scribbleBodyMedium)
                    .fontWeight(.bold)
                    .foregroundStyle(Color.scribblePrimary)

                TrendBadge(direction: direction)
            }
        }
    }
}

private struct TrendBadge: View {
    let direction: TrendDirection

    var body: some View {
        let (text, color) = {
            switch direction {
            case .improving: return (String(localized: "IMPROVING"), Color.scribbleSuccess)
            case .stable: return (String(localized: "STABLE"), Color.scribbleMidGray)
            case .plateaued: return (String(localized: "PLATEAUED"), Color.scribbleWarning)
            case .declining: return (String(localized: "DECLINING"), Color.scribbleError)
            }
        }()

        Text(text)
            .font(.system(size: 10, weight: .bold))
            .foregroundStyle(color)
            .padding(.horizontal, 10)
            .padding(.vertical, 4)
            .background(color.opacity(0.1))
            .clipShape(Capsule())
    }
}
